hadoop-hdds/common/src/main/java/org/apache/hadoop/hdds/security/ssl/ReloadingX509TrustManager.java

```diff
@@ -127,23 +129,22 @@ public ReloadingX509TrustManager loadFrom(CertificateClient caClient) {
   X509TrustManager loadTrustManager(CertificateClient caClient)
       throws GeneralSecurityException, IOException {
     // SCM certificate client sets root CA as CA cert instead of root CA cert
-    X509Certificate rootCACert = caClient.getRootCACertificate() == null ?
-        caClient.getCACertificate() : caClient.getRootCACertificate();
+    Set<X509Certificate> rootCACerts = caClient.getAllRootCaCerts().isEmpty() ?
+        caClient.getAllCaCerts() : caClient.getAllRootCaCerts();
 
-    String rootCACertId = rootCACert.getSerialNumber().toString();
     // Certificate keeps the same.
-    if (currentRootCACertId != null &&
-        currentRootCACertId.equals(rootCACertId)) {
+    if (rootCACerts.size() > 0 &&
+        currentRootCACertIds.size() == rootCACerts.size() &&
+        !rootCACerts.stream().filter(
+            c -> !currentRootCACertIds.contains(c.getSerialNumber().toString()))
+            .findAny().isPresent()) {
```

Comment:
```suggestion
       rootCACerts.stream().allMatch(c ->
            currentRootCACertIds.contains(c.getSerialNumber().toString()))) {
```

Reply:
Also this we might change in the ReloadingX509KeyManager at L240-242 (loadKeyManager method).

Reply:
Nice suggestion.